addons/sourcemod/scripting/ConnectAnnounce.sp

```diff
@@ -88,6 +112,109 @@ public void OnPluginStart()
 //----------------------------------------------------------------------------------------------------
 // Purpose:
 //----------------------------------------------------------------------------------------------------
+public void OnAllPluginsLoaded()
+{
+#if defined _EntWatch_include
+	g_bEntWatch = LibraryExists("EntWatch");
+#endif
+
+#if defined _KnockbackRestrict_included_
+	g_bKbRestrict = LibraryExists("KnockbackRestrict");
+#endif
+
+#if defined _sourcebanschecker_included
+	g_bSbChecker = LibraryExists("sourcebans++");
+#endif
+
+	VerifyNatives();
+}
+
+public void OnLibraryRemoved(const char[] name)
+{
+	HandleLibraryChange(name, false);
+}
+
+public void OnLibraryAdded(const char[] name)
+{
+	HandleLibraryChange(name, true);
+}
+
+void HandleLibraryChange(const char[] name, bool isAdded = false)
+{
+	#if defined _Connect_Included
+	if (strcmp(name, "connect.ext", false) == 0)
+	{
+		VerifyNative_Connect();
+	}
+	#endif
+	#if defined _EntWatch_include
+	if (strcmp(name, "EntWatch", false) == 0)
+	{
+		g_bEntWatch = isAdded;
+		VerifyNative_EntWatch();
+	}
+	#endif
+	#if defined _KnockbackRestrict_included_
+	if (strcmp(name, "KnockbackRestrict", false) == 0)
+	{
+		g_bKbRestrict = isAdded;
+		VerifyNative_KbRestrict();
+	}
+	#endif
+	#if defined _sourcebanschecker_included
+	if (strcmp(name, "sourcebans++", false) == 0)
+	{
+		g_bSbChecker = isAdded;
+		VerifyNative_SbChecker();
+	}
+	#endif
+}
+
+stock void VerifyNatives()
+{
+	VerifyNative_Connect();
+	VerifyNative_EntWatch();
+	VerifyNative_KbRestrict();
+	VerifyNative_SbChecker();
+}
+
+stock void VerifyNative_Connect()
+{
+	#if defined _Connect_Included
+	char sError[255];
+	int iStatus = GetExtensionFileStatus("connect.ext", sError, sizeof(sError));
```

Comment:
why do you need this

Reply:
Because extension never call `OnAllPluginsLoaded`

Reply:
They also do not call `OnLibraryAdded` on their first load, but only on late load.